packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -60,6 +60,11 @@ class AnalyzeCommand extends FlutterCommand {
         negatable: false,
         hide: !verboseHelp,
         help: 'Also output the analysis time.');
+    // Hidden option to allow mixed non-nullable support.
+    argParser.addFlag('permit-non-nullable',
```

Comment:
Could you expand on either the comment or the help text to explain when this is needed?

Reply:
I added more comment here, also explaining below:

This is only required because we analyze the repo all at once, so I can opt-in more granularly. This flag can go away once flutter starts using null safety features